crypto/stark-verifier-ethereum/contracts/default_cs.sol

```diff
@@ -59,42 +53,55 @@ abstract contract DefaultConstraintSystem is ConstraintSystem, Trace  {
         for (uint256 i = 0; i < queries.length; i++) {
             uint256 result = 0;
             {
+            trace('get_polynomial_points_loop_1', true);
             // These held pointers help soldity make the stack work
             uint256[] memory trace_oods_value = data.trace_oods_values;
             uint256[] memory trace_values = data.trace_values;
-            for (uint256 j = 0; j < trace_oods_value.length; j++) {
+            for (uint256 j = 0; j < trace_oods_value.length; ) {
                 uint256 loaded_trace_data = trace_oods_value[j];
                 // J*2 is the col index when the layout is in coloum major form
                 // NUM_COLUMNS*i idenifes the start of this querry's row values
                 uint256 calced_index = NUM_COLUMNS*i + layout[j*2];
-                uint256 numberator = trace_values[calced_index].fsub(loaded_trace_data);
+                uint256 numberator = addmod(trace_values[calced_index], (PrimeField.MODULUS - loaded_trace_data), PrimeField.MODULUS);
 
                 // Our trace layout is: (Col, Row Inverse Index),
                 // So the following will tell us where to look in the inverses
                 uint256 row = layout[j*2+1];
                 calced_index = (NUM_OFFSETS+1)*i + row;
                 uint256 denominator_inv = inverses[calced_index];
 
-                uint256 element = numberator.fmul(denominator_inv);
+                uint256 element = mulmod(numberator, denominator_inv, PrimeField.MODULUS);
                 uint256 coef = coeffiecients.next();
-                uint256 next_term = element.fmul_mont(coef);
-                result = result.fadd(next_term);
+                uint256 next_term = mulmod(mulmod(element, coef, PrimeField.MODULUS), PrimeField.MONTGOMERY_R_INV, PrimeField.MODULUS);
+                result = addmod(result, next_term, PrimeField.MODULUS);
+
+                assembly {
+                    j := add(j, 1)
+                }
             }
+            trace('get_polynomial_points_loop_1', false);
+
             }
 
+            trace('get_polynomial_points_loop_2', true);
             uint256 denominator_inv = inverses[i * (NUM_OFFSETS+1) + NUM_OFFSETS];
             uint256 len = CONSTRAINT_DEGREE;
             uint256[] memory constraint_values = data.constraint_values;
             uint256[] memory constraint_oods_values = data.constraint_oods_values;
-            for (uint256 j = 0; j < len; j++) {
+            for (uint256 j = 0; j < len; ) {
```

Comment:
It seems crazy that this is more gas-efficient.

Reply:
Lol yeah it saves around 50k gas, not so much but still pretty significant. Our hottest loops run so much that each gas is super valuable